src/Infrastructure.Dapper/Vault/Repositories/CipherRepository.cs

```diff
@@ -562,6 +562,57 @@ public async Task CreateAsync(IEnumerable<Cipher> ciphers, IEnumerable<Folder> f
                         }
                     }
 
+                    if (existingFolders.Any())
+                    {
+
+                        // 1. Create temp tables to bulk copy into.
+
+                        var sqlCreateTemp = @"
+                            SELECT TOP 0 *
+                            INTO #TempFolder
+                            FROM [dbo].[Folder]";
+
+                        using (var cmd = new SqlCommand(sqlCreateTemp, connection, transaction))
+                        {
+                            cmd.ExecuteNonQuery();
+                        }
+
+                        // 2. Bulk copy into temp tables.
+                        using (var bulkCopy = new SqlBulkCopy(connection, SqlBulkCopyOptions.KeepIdentity, transaction))
+                        {
+                            bulkCopy.DestinationTableName = "#TempFolder";
+                            var dataTable = BuildFoldersTable(bulkCopy, existingFolders);
+                            bulkCopy.WriteToServer(dataTable);
+                        }
+
+                        // 3. Insert into real tables from temp tables and clean up.
+
+                        var sql = string.Empty;
+
+                        sql += @"
+                            UPDATE
+                                [dbo].[Folder]
+                            SET
+                                [Name] = TF.[Name],
+                                [RevisionDate] = TF.[RevisionDate]
+                            FROM
+                                [dbo].[Folder] F
+                            INNER JOIN
+                                #TempFolder TF ON F.Id = TF.Id
+                            WHERE
+                                F.[UserId] = @UserId";
+
+
+                        sql += @"
+                            DROP TABLE #TempFolder";
```

Comment:
Use `StringBuilder` if you have to build strings this way, but I don't see why it's necessary. You could just initialize all of this in the declaration.